### ISSUE TITLE
grunt clean does not clean generated folder

### ISSUE THREAD

**Issue by gwharton** [magento/magento2#25739]
### Preconditions (*)
1. Magento 2.3-develop
2. Copy the default grunt config

```
  cp grunt-config.json.sample grunt-config.json
  cp Gruntfile.js.sample Gruntfile.js
```

### Steps to reproduce (*)
1. Enable production mode
2. php bin/magento setup:upgrade
3. php bin/magento setup:di:compile
4. php bin/magento setup:static-content:deploy
5. grunt clean

### Expected result (*)
1. generated/code should be empty
2. generated/metadata should be empty

### Actual result (*)
1. They are not empty.
2. Analysis of grunt config shows it is still trying to clear var/generation/**/*

**Comment by andrewbess** [comment]
@magento I am working on it

**Comment by magento-engcom-team** [comment]
@Stepa4man Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [x] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by VladimirZaets** [comment]
Hi @gwharton. Thank you for your report. 
The issue has been fixed in magento/magento2#25765 by @andrewbess in 2.4-develop branch
Related commit(s):
 - [b9ef877799ab4e94e5d5fd5a50342f7c8c8c8c6a](https://github.com/magento/magento2/commit/b9ef877799ab4e94e5d5fd5a50342f7c8c8c8c6a)

The fix will be available with the upcoming 2.4.0 release.
